### ISSUE TITLE
--ignore-modules fix bad formatting

### ISSUE THREAD

**Issue by LiamSkirrow** [LiamSkirrow/verilogtree#39]
Including the --ignore-modules flag can mess around with the formatting a little, leading to some weird output.

To reproduce: `./verilogtree -f rtl/simple/*.v --no-inst-name -n mod1`
Gives the output:
```
altTop
└── mod2  

top
├── mod0  
│   ├── mod3  
│   │   └── mod4  
└── mod2  

Ignoring following module(s) from output: mod1
```

**Comment by LiamSkirrow** [comment]
The implementation of this may be made easier by simply pre-processing the tree a little and pruning out the --ignore-module modules. This means the output text formatting function can be left completely untouched and it just has to operate on the modified/pruned tree.

**Comment by LiamSkirrow** [comment]
The scope of the feature is being changed (lol). The original intention of the `--ignore-modules` flag was to list module names that are to be ignored in the output. However, the current way I've implemented the recursive tree construction relies mainly on referencing child nodes, which makes things tricky. 

Instead, now the --ignore-modules option will ignore the CHILDS of the supplied modules. For example, going `--ignore-modules mod3` will give:

```
top
├── mod0
│   ├── mod3
│   └── mod1
│       └── mod3
├── mod1
│   └── mod3
└── mod2
    └── mod1
        └── mod3

altTop
└── mod2
    └── mod1
        └── mod3
```

Note how mod3 is still shown, but its children have been minimised...

**Comment by LiamSkirrow** [comment]
Reopened since I've just realised that we can't minimise the childs of the tree roots. This would be useful so should be implemented.

For example, if `-n top` is supplied, then the top level module childs should be removed from the output...

**Comment by LiamSkirrow** [comment]
Now fixed in merge commit `93785d8f`, simple fix to remove the children of top level modules manually by clearing the childNode vector.
